packages/terra-application-layout/docs/usage.md

```diff
@@ -0,0 +1,264 @@
+        }
+      }
+    }
+  }
+}
+```
+
+If the global router location is `/a/b/c`, the ApplicationLayout will render `CMenuComponent` in the menu region and `AContentComponent` in the content region. However, given the nested nature of the configuration, the `BMenuComponent` and `AMenuComponent` may still provide important navigation or contextual information for the `AContentComponent`. It would be nice to be able to be able to view those menus without changing the global router location and losing the current users current application context.
+
+The `routingStackDelegate` can do this. When the ApplicationLayout determines (based on path segment comparisons) that a component preceeds the currently presented component, the presented component's `routingStackDelegate` prop will have a defined `showParent` value. When called, `showParent` will change the local location of that stack of components to be that preceeding path. For this example, calling `showParent` from `CMenuComponent` will cause `BMenuComponent` to be displayed. However, the global router location will not change and remain `/a/b/c`. This way, if `AContentComponent` was rendering some content differently based on the additional paths, calling `showParent` will not impact it at all.
+
+Assume `showParent` was selected from `CMenuComponent`, resulting in the presentation of `BMenuComponent`, `showParent` could be called again from within `BMenuComponent`, resulting in the rendering of `AMenuComponent` and a local location of `/a`. The global router location will still be `/a/b/c`, and `AContentComponent` will be oblivious to the changes happening in the menu beside it. Since the last component in the hierarchy has been reached, `showParent` will not be defined on the `AMenuComponent`'s `routingStackDelegate`.
+
+At that point, the user may want to return to their original state and view the `CMenuComponent` again. However, this can be difficult. If the `AMenuComponent` links directly to the `BMenuComponent`, the link will change the global router location to `/a/b` and the user's previous context would be lost. Therefore, `AMenuComponent` should use the other API on the `routingStackDelegate` called `show`. The `show` function will intelligently route to the location provided to it. If the path provided to `show` matches the global router location, or if is a subset match of the global router location, `show` will update the local location to that new route, allowing for virtual downward traversal through the menu stack. If a match to the global router location cannot be determined, `show` will then and only then update the global router location for the new value. Thus, if `show('/a/b')` is called from `AMenuComponent`, `BMenuComponent` will be presented, while the global router context remains `/a/b/c`. If `BMenuComponent` were to then call `show('/a/b/c')`, `CMenuComponent` would be presented, and the local location would again match the global one.
```

Comment:
and the local location would again match the global one. Lets define the new global for added clarity.

Reply:
With the addition of the RoutingMenu, I think this admittedly confusing example can be removed. I'll be adding RoutingMenu documentation that will aim to explain the same concepts more concisely.